### ISSUE TITLE
YouTube Transcripts

### ISSUE THREAD

**Issue by nkandpa2** [r-three/common-pile#53]
Videos on YouTube can optionally be published under a CC-BY license. We can identify these videos with the YouTube API, download them, and transcribe them with an ASR system like [whisperX](https://github.com/m-bain/whisperX).

**Comment by nkandpa2** [comment]
Code for cataloging CC YouTube videos can be found in this repo: https://github.com/nkandpa2/youtube-commons. About 800K CC videos adding up to about 300K hours of primarily speech-based video have been cataloged.

**Comment by craffel** [comment]
Can we find any explicit statement that these videos are public domain, or do we need to rely on the reasoning that "these were made by the government, therefore they are public domain"

**Comment by storytracer** [comment]
No, I was not able to find any explicit statements inside the government YouTube channels so far that the videos they host are in the public domain except by using the full-text search to search for "Public Domain" which only finds 3,000 videos: https://find.search.gov/search/news?affiliate=usagov_all_gov&channel=10448&sort_by=&query=%22public+domain%22.

The only explicit statement about a channel I was able find so far is on the USDA website, which states that all the content in their YouTube channel is in the public domain in a [privacy assessment](https://www.usda.gov/sites/default/files/documents/oc_youtube_pia.pdf) (p. 4): "Video content published to the USDA YouTube page will be previously approved by relevant Department and Office of Communications leadership, and will be available in the public domain." But while USDA makes this statement in this random document on their website, they do not include this statement in their YouTube channel or the videos themselves, which are indexed through find.search.gov. 

Generally speaking, government videos are rarely expliclity declared to be in the public domain. But that's the standard case for federal government documents as well. Outside the cultural heritage sector works are rarely explicitly declared to be in the public domain with something like the [PD Mark](https://creativecommons.org/publicdomain/mark/1.0/deed.en) and even in the CH sector the use of a PD mark is much less common than CC licenses.

US Federal Government works are ["born in the public domain"](https://web.law.duke.edu/cspd/publicdomainday/2024/#_ftn10) [by law](https://www.copyright.gov/title17/92chap1.html#105), they don't become part of the public domain at some later point through a waiver or expiration of rights. But such a general legal status is of course harder to document than an explicit rights statement, so it's worth evaluating whether it's worth the effort to transcribe government videos in addition to CC videos.

**Comment by nkandpa2** [comment]
This is a great idea @storytracer. I think we can cover many of these videos by simply including the major US agencies' YouTube channels in the dataset. Looking through a couple of them from the search results you provided, channels sometimes will explicitly state in the description something along the lines of "Here you will find original content produced by [AGENCY]". So to me it feels reasonable to count this data as PD. We can always remove this later if we are unsure of the PD status.

**Comment by storytracer** [comment]
@nkandpa2 What's the current state of cataloging? Do you have an ETA and do think you could upload the catalog as a dataset to HF? I would like to help with the distributed processing of the audio files on our infrastructure!
